src/include/stir/FilePath.inl

```diff
@@ -0,0 +1,70 @@
+/*
+    Copyright (C) 2018, University of Hull
+
+    This file is part of STIR.
+    This file is free software; you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation; either version 2.1 of the License, or
+    (at your option) any later version.
+
+    This file is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+
+    See STIR/LICENSE.txt for details
+*/
+
+/*!
+  \file
+  \ingroup buildblock
+
+  \brief Implementations of inline functions for class stir::FilePath
+
+  \author Nikos Efthimiou
+  \author Kris Thielemans
+  \author PARAPET project
+
+*/
+START_NAMESPACE_STIR
+
+inline bool FilePath::operator==(const FilePath& other) {
+    if (this->my_string==other.my_string)
+        return true;
+    else
+        return false;
+}
+
+inline bool FilePath::operator==(const std::string& other) {
+    if (this->my_string==other)
+        return true;
+    else
+        return false;
+}
+
+inline void FilePath::operator=(const FilePath& other) {
+    this->my_string = other.my_string;
+    this->separator = other.separator;
+}
+
+//! \warning Possibly dangerous... when real paths are used.
```

Comment:
I really don't understand what's dangerous here... sorry.

Reply:
I the case that we have a real path, set during construction, that passes tests like exists and then with the  assignment we set a fake string. If we don't repeat checks (e.g. the one I added about the separator) we can run into trouble.

Reply:
ok. I think you need to document what the checks do (I get the impression that setting the flag at constructor time means you will presume checks need to be done always?), that `append` will create the dir (kind of non-obvious from the name if you ask me). and then add something like the above in the doc as well. (By the way, why not run the check then here?)

Reply:
Well, the initial notion was to run the check on the constructor. Then I saw the potential problem and I added the checks pretty much everywhere needed. Maybe a bit redundant now that I think about it. 
Append is the command in both boost::filesystem and C17